Podcast/SearchEndpointRequests.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  SearchEndpointRequests.swift
+//  Podcast
+//
+//  Created by Jack Thompson on 8/27/18.
+//  Copyright © 2018 Cornell App Development. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+class SearchEndpointRequest: EndpointRequest {
+    
+    var query: String!
+    var offset: Int!
+    var max: Int!
+    
+    required init(modelPath: String = "", query: String, offset: Int = 0, max: Int = 0) {
+        super.init()
+        
+        path = "/search/\(modelPath)/\(query.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) ?? query)/"
+        
+        httpMethod = .get
+        
+        queryParameters = ["offset": offset, "max": max]
+    }
+}
+
+class SearchAllEndpointRequest: SearchEndpointRequest {
+    
+    required init(modelPath: String = "all", query: String, offset: Int, max: Int) {
+        super.init(modelPath: modelPath, query: query, offset: offset, max: max)
+    }
+    
+    override func processResponseJSON(_ json: JSON) {
+        let episodes = json["data"]["episodes"].map{ episodeJSON in
+            Cache.sharedInstance.update(episodeJson: episodeJSON.1)
+        }
+        let series = json["data"]["series"].map{ seriesJSON in
+            Cache.sharedInstance.update(seriesJson: seriesJSON.1)
+        }
+        let users = json["data"]["users"].map{ userJSON in
+            Cache.sharedInstance.update(userJson: userJSON.1)
+            }.filter{
```

Comment:
space after filter and before the bracket `.filter {`

Reply:
Should we do this for all the maps above as well?

Reply:
Yeah, I believe swiftlint might automatically fix this